Basic-Video-Call/js/app.js

```diff
@@ -0,0 +1,138 @@
+/* global OT API_KEY TOKEN SESSION_ID SAMPLE_SERVER_BASE_URL */
+
+var apiKey;
+var sessionId;
+var token;
+
+// Handling all of our errors here by alerting them
+function handleError(error) {
+  if (error) {
+    alert(error.message);
+  }
+}
+
+
+function toggleStyle(button) { // toggles button css class
+  button.classList.toggle('toggle-button-on');
+  button.classList.toggle('toggle-button-off');
+}
+
+
+function initializeSession() {
+  var session = OT.initSession(apiKey, sessionId);
+
+  // Subscribe to a newly created stream
+  session.on('streamCreated', function (event) {
+    session.subscribe(event.stream, 'subscriber', {
+      insertMode: 'append',
+      width: '100%',
+      height: '100%'
+    }, handleError);
+  });
+
+  // Create a publisher
+  var publisher = OT.initPublisher('publisher', {
+    insertMode: 'append',
+    width: '100%',
+    height: '100%'
+  }, handleError);
+
+  // If you disconnect from a call, overrride the default behaviour.
+  publisher.on('streamDestroyed', function (event) {
+    if (session.connection === null || event.stream.connection.connectionId === session.connection.connectionId) {
+      event.preventDefault();
+    }
+  });
+
+  // Connect to the session
+  session.connect(token, function (error) {
+    // If the connection is successful, publish to the session
+    if (error) {
+      handleError(error);
+    } else {
+      session.publish(publisher, handleError);
+    }
+  });
+
+  function toggleMicrophone() {
+    if (publisher.session) publisher.publishAudio(!publisher.stream.hasAudio);
+    var microphoneToggle = document.getElementById('microphone-toggle');
+    toggleStyle(microphoneToggle);
+  }
```

Comment:
Maybe extract this out of the initializeSession() ?

Reply:
now extracted